### PR TITLE
feat(udf): add metric of UDF memory usage

### PR DESCRIPTION
**Pull request by wangrunji0408**
Signed-off-by: Runji Wang <wangrunji0408@163.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR adds a metric of UDF memory usage. Currently only embedded JS UDF supports that, others will always show 0. This PR also replaces UDF metrics by our label-guarded version.

metric-memory-usage

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by chenzl25 — APPROVED** [review]
LGTM!

File: grafana/risingwave-dev-dashboard.dashboard.py [hunk 1]
```diff
@@ -4268,6 +4268,20 @@ def section_udf(outer_panels):
                         ),
                     ],
                 ),
+                panels.timeseries_bytes(
+                    "UDF Memory Usage (bytes)",
+                    "",
```
Comment (xxchan) [inline grafana/risingwave-dev-dashboard.dashboard.py:4273]:
```suggestion
                    "Currently only embedded JS UDF supports this. Others will always show 0.",
```

File: src/expr/core/src/expr/expr_udf.rs [hunk 2]
```diff
@@ -328,18 +341,33 @@ impl MetricsVec {
...
+    fn with_label_values(
+        &self,
+        link: &str,
+        language: &str,
+        identifier: &str,
+        fragment_id: &str,
+    ) -> Metrics {
+        // generate an unique id for each instance
+        static NEXT_INSTANCE_ID: AtomicU64 = AtomicU64::new(0);
+        let instance_id = NEXT_INSTANCE_ID.fetch_add(1, Ordering::Relaxed).to_string();
```
Comment (xxchan) [inline src/expr/core/src/expr/expr_udf.rs:357]: Why is this needed?

Comment (wangrunji0408) [reply src/expr/core/src/expr/expr_udf.rs:357]: The memory usage is a gauge. So we need to distinguish each instance's metric by a different label, otherwise the value will be overridden rather than accumulated.

**Review by xxchan — APPROVED** [review]
